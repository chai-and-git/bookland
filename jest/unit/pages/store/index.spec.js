import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueMeta from 'vue-meta';
import index from '@/pages/store/index.vue';

const localVue = createLocalVue();
localVue.use(VueMeta, { keyName: 'head' });

jest.mock("axios", () => ({
  get: () => Promise.resolve({ data: [{
    "books": [
      {
        "book_id": "1",
        "name": "The Alchemist",
        "author": "Paulo Coelho",
        "description": "Paulo Coelho's enchanting novel has inspired a devoted following around the world. This story, dazzling in its powerful simplicity and inspiring wisdom, is about an Andalusian shepherd boy named Santiago who travels from his homeland in Spain to the Egyptian desert in search of a treasure buried in the Pyramids. Along the way he meets a Gypsy woman, a man who calls himself",
        "price": 210,
        "imagePath_small": "require('@/assets/book-alchemist.jpg')",
        "quantity": 1
      }
    ]
  }]})
}));

const indexWrapper = shallowMount(index, { localVue });

describe('index.vue', () => {
  it('is a Vue instance', () => {
      const wrapper = indexWrapper;
      expect(wrapper.vm).toBeTruthy();
  })

  it('is a Vue instance', () => {
    const wrapper = indexWrapper;
    expect(wrapper.html()).toMatchSnapshot();
  })
})

describe("index.vue", () => {
  it("mocking the axios call to get books from axios", () => {
    var wrapper = indexWrapper;
    expect(wrapper.vm.books.length).toBe(1);
  });
});