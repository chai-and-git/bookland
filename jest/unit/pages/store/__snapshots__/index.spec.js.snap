// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.vue is a Vue instance 1`] = `
"<b-jumbotron-stub headertag=\\"h1\\" headerlevel=\\"3\\" leadtag=\\"p\\" tag=\\"div\\" id=\\"jumbotron\\">
  <h1 id=\\"book-category\\">All Books</h1>
  <b-list-group-stub tag=\\"div\\" id=\\"book-list\\">
    <b-list-group-item-stub tag=\\"div\\" target=\\"_self\\" id=\\"book-list-item\\">
      <b-container-stub tag=\\"div\\" id=\\"container\\">
        <b-row-stub tag=\\"div\\" id=\\"container-row\\">
          <b-col-stub cols=\\"12\\" sm=\\"auto\\" tag=\\"div\\" id=\\"container-image-col\\">
            <b-img-stub blankcolor=\\"transparent\\"></b-img-stub>
          </b-col-stub>
          <b-col-stub tag=\\"div\\" id=\\"container-book-detail-col\\">
            <div>
              <h3></h3>
            </div>
            <div id=\\"book-author-row\\">
              <h4></h4>
            </div>
            <div id=\\"book-desc-row\\">
              <p id=\\"book-desc\\"></p>
            </div>
            <div id=\\"book-price-row\\">
              <p id=\\"book-price\\">Â£ </p>
            </div>
            <div id=\\"add-to-cart-row\\">
              <b-button-stub variant=\\"outline-secondary\\" type=\\"button\\" tag=\\"button\\" target=\\"_self\\">Add to Cart</b-button-stub>
            </div>
          </b-col-stub>
        </b-row-stub>
      </b-container-stub>
    </b-list-group-item-stub>
  </b-list-group-stub>
</b-jumbotron-stub>"
`;
